invenio_rdm_records/services/services.py

```diff
@@ -128,3 +141,282 @@ def oai_result_item(self, identity, oai_record_source):
             record,
             links_tpl=self.links_item_tpl,
         )
+
+    #
+    # Deletion workflows
+    #
+    @unit_of_work()
+    def delete_record(self, identity, id_, data, expand=False, uow=None):
+        """(Soft) delete a published record."""
+        record = self.record_cls.pid.resolve(id_)
+        if record.deletion_status.is_deleted:
```

Comment:
No, this shouldn't be the case as we're specifically resolving a `record` here.
In the generator, I went a bit more defensive because I think the `record` there could be an actual **draft** as well (and they don't have a `deletion_status`).